packages/rich-text/src/apply-format.js

```diff
@@ -60,19 +70,23 @@ export function applyFormat(
 		}
 	} else {
 		for ( let index = startIndex; index < endIndex; index++ ) {
-			applyFormats( newFormats, index, format );
+			if ( newFormats[ index ] ) {
+				newFormats[ index ] = newFormats[ index ]
+					.filter( ( { type } ) => type !== format.type );
+			} else {
+				newFormats[ index ] = [];
+			}
 		}
-	}
 
-	return normaliseFormats( { ...value, formats: newFormats } );
-}
+		const position = Math.min(
```

Comment:
I've been staring at this for a while trying to understand, and am having trouble grasping why it matters the number of formats at the start or end of the fragment of text. Should it not instead be some result of whichever format is applied to the broadest selection? Or is that an indirect result of this logic?

Reply:
Yes, you're right, it doesn't account for the following case:

```
22  33
111111
```

where you'd expect the format to be applied as follows:

```
22  33
444444
111111
```

Reply:
Fixed in a55ce96. Thanks!